src/lazy/binary/raw/v1_1/immutable_buffer.rs

```diff
@@ -1544,7 +1541,8 @@ mod tests {
 
         // Construct an encoding directive that defines this number of macros. Each macro will expand
         // to its own address.
-        let mut macro_definitions = String::from("$ion_encoding::(\n  (macro_table\n");
+        let mut macro_definitions =
+            String::from("$ion_encoding::(\n  (macro_table $ion_encoding\n");
```

Comment:
🗺️ This PR modified the encoding directives to create a new macro table each time while also supporting re-exports of `$ion` and `$ion_encoding`. This meant that several unit tests that used multiple encoding directives saw macro definitions disappear. I've modified those tests to re-export `$ion_encoding` (that is: do a macro table append) as needed.